File: sdks/go/corthan/organisation.go
package corthan

import (
	"context"
	"fmt"
	"net/http"
)

// OrganisationService wraps Organisation operations.
type OrganisationService struct {
	t *TransportClient
}

// List retrieves organizations.
func (s *OrganisationService) List(ctx context.Context) (*OrganizationListSuccess, error) {
	var result OrganizationListSuccess
	err := s.t.Request(ctx, http.MethodGet, "/organizations", nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// Create registers a new organization.
func (s *OrganisationService) Create(ctx context.Context, org *Organization) (*OrganizationSuccess, error) {
	var result OrganizationSuccess
	err := s.t.Request(ctx, http.MethodPost, "/organizations", org, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// ListMembers retrieves user memberships of an organization.
func (s *OrganisationService) ListMembers(ctx context.Context, id string) (*OrganizationMemberListSuccess, error) {
	path := fmt.Sprintf("/organizations/%s/members", id)
	var result OrganizationMemberListSuccess
	err := s.t.Request(ctx, http.MethodGet, path, nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: sdks/node/src/services/organisation.ts
import { TransportClient } from "../transport";
import { Organization, OrganizationListSuccess, OrganizationMemberListSuccess, OrganizationSuccess } from "../types";

/**
 * Service for managing Organizations and multi-tenant domain mapping.
 */
export class OrganisationService {
  constructor(private transport: TransportClient) {}

  /**
   * Registers a new business tenant Organization.
   */
  public async create(org: Organization): Promise<OrganizationSuccess> {
    return this.transport.request<OrganizationSuccess>("POST", "/organizations", org);
  }

  /**
   * Retrieves organization configuration by identifier.
   */
  public async get(id: string): Promise<OrganizationSuccess> {
    return this.transport.request<OrganizationSuccess>("GET", `/organizations/${id}`);
  }

  /**
   * Lists all organization tenants mapped to the authenticated user.
   */
  public async list(): Promise<OrganizationListSuccess> {
    return this.transport.request<OrganizationListSuccess>("GET", "/organizations");
  }

  /**
   * Lists users who are members of an organization.
   */
  public async listMembers(orgId: string): Promise<OrganizationMemberListSuccess> {
    return this.transport.request<OrganizationMemberListSuccess>("GET", `/organizations/${orgId}/members`);
  }
}
